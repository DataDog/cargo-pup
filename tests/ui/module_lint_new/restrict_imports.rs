// This product includes software developed at Datadog (https://www.datadoghq.com/) Copyright 2024 Datadog, Inc.

//@compile-flags: --crate-name test_restrict_imports
//@compile-flags: --crate-type lib

// Define dummy modules to be used for imports
pub mod dummy_denied {
    pub fn anything() {}
}

// Allowed imports (std and core are in allowed list)
use std::fmt;
use std::io;
use core::ops::Add;

// Disallowed import (not in allowed list) - will trigger both rules
use crate::dummy_denied::anything; //~ ERROR: Use of module 'crate::dummy_denied::anything' is not allowed; only ["std::*", "core::*"] are permitted
                                  //~^ ERROR: Use of module 'crate::dummy_denied::anything' is denied

// Nested modules with allowed imports
pub mod nested {
    // These should also be checked against the rules
    use std::collections::HashMap; // Allowed - in allowed list
} 
