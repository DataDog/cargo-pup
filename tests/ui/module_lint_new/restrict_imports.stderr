error: Use of module 'crate::dummy_denied::anything' is not allowed; only ["std::*", "core::*"] are permitted
  --> tests/ui/module_lint_new/restrict_imports.rs:15:1
   |
LL | use crate::dummy_denied::anything;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Use only allowed module imports
   = note: Applied by cargo-pup rule 'restrict_imports_allowed_test'.
   = note: `#[deny(module_restrict_imports)]` on by default

error: Use of module 'crate::dummy_denied::anything' is denied
  --> tests/ui/module_lint_new/restrict_imports.rs:15:1
   |
LL | use crate::dummy_denied::anything;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Remove this import
   = note: Applied by cargo-pup rule 'restrict_imports_denied_test'.

error: aborting due to 2 previous errors

